lib/openstack_query/query_blocks/query_output.py

```diff
@@ -154,9 +176,28 @@ def generate_output(
         :param openstack_resources: List of openstack objects to obtain properties from - e.g. [Server1, Server2]
         :return: List containing dictionaries of the requested properties obtained from the items
         """
-        return [self._parse_property(item) for item in openstack_resources]
+        output = []
+        for item in openstack_resources:
+            prop_list = self._parse_properties(item)
+            if self.forwarded_outputs:
+                for grouped_property, outputs in self.forwarded_outputs.items():
+                    prop_val = self._parse_property(grouped_property, item)
+
+                    # this "should not" error because forwarded outputs should always be a super-set
+                    # but sometimes resolving the property fails for whatever reason
+                    # in this case just copy all the keys in one of the dictionaries in the list
+                    # since they should be equal and set the values to "Not Found"
+                    try:
+                        output_list = outputs[prop_val]
+                        # should only ever contain one element since it's grouped by a unique id
+                        prop_list.update(output_list[0])
+                    except KeyError:
+                        prop_list.update({key: "Not Found" for key in outputs.keys()})
```

Comment:
Could we not fail noisy? I can see `Not Found` keys being really confusing for end-users

Reply:
Sure I guess?

This happens when there's a mismatch between the way we get properties between two link properties - this may be pretty common or pretty rare - I don't know as of yet

Reply:
Can we make it noisy and obvious to figure out if it is common or rare
We can make it more permissive in the future by re-adding this, but we can't easily restrict it once people rely on this